Remove SDS refresh timeout

adiprerepa opened the pull request:
[ X] Networking
[X] Security

Fixes #25339. Removes the `initial_fetch_timeout` so we wait indefinitely for the cluster's secret.

adiprerepa committed:
init

adiprerepa committed:
init

adiprerepa committed:
fix test

adiprerepa committed:
fix test

istio-testing commented:
@adiprerepa: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | aefbbf1c8c4049721f5a9fcd1c50e0ef92010609 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/lint_istio/15635) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

adiprerepa committed:
lint

adiprerepa committed:
lint

istio-testing commented:
@adiprerepa: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | fd4bb83e012589e38106cdee5caf8227c2acbd46 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/unit-tests_istio/16230) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

Commit:
istiocoredns chart should not depend on global.yaml (#25386)

howardjohn commented on pilot/pkg/security/model/authentication.go:
```diff
@@ -125,7 +126,7 @@ func ConstructSdsSecretConfig(name, sdsUdsPath string) *tls.SdsSecretConfig {
 				},
 			},
-			InitialFetchTimeout: features.InitialFetchTimeout,
+			InitialFetchTimeout: ptypes.DurationProto(time.Second * 0),
```
nit: warrants a comment why we do *not* follow `features.InitialFetchTimeout` but still do in the other usages. Specifically, that we want it for gateway certs which may or may not exist, but not for workload certs which must exist

JimmyCYJ replied:
@adiprerepa please add a brief explanation above this line as @howardjohn commented.

lambdai replied:
I am not sure if I full understand it
Is that listener at ingressgateway should set features.InitialFetchTimeout while cluster at ingressgateway(and for all sidecars) should set infinite initialfetchtimeout?

costinm replied:
+1 - infinite doesn't sound good.. Comment explaining the problem, solution would help people who'll work on this in the future.

howardjohn replied:
If we are using TLS I would rather a pod fail to startup then to startup without certs and suddenly lead to random traffic failures. 100% agree on comments though

adiprerepa replied:
sure ill add the comment

adiprerepa replied:
this is going to take a while, so many tests fail with this change.

JimmyCYJ replied:
Is there a ETA for this? It would be great if this could be shipped with OSS 1.7. https://github.com/istio/istio/wiki/Istio-Release-1.7#timeline

adiprerepa replied:
ill try to get this shipped @JimmyCYJ by 1.7

Commit:
Remove prometheus chart dependency on global.yaml (#25370)

* Remove prometheus chart dependency on global.yaml

* Add meshConfig.enablePrometheusMerge value

adiprerepa committed:
test hardcode

adiprerepa committed:
test hardcode

adiprerepa committed:
revert stupid commit

adiprerepa committed:
revert stupid commit

istio-testing commented:
@adiprerepa: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | fd4bb83e012589e38106cdee5caf8227c2acbd46 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/unit-tests_istio/16230) | `/test unit-tests_istio`
lint_istio | 58f508587be7eab1419862ba4f5df26f4d8e05b9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/lint_istio/15654) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@adiprerepa: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | fd4bb83e012589e38106cdee5caf8227c2acbd46 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/unit-tests_istio/16230) | `/test unit-tests_istio`
lint_istio | 58f508587be7eab1419862ba4f5df26f4d8e05b9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/lint_istio/15654) | `/test lint_istio`
gencheck_istio | 58f508587be7eab1419862ba4f5df26f4d8e05b9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/gencheck_istio/14249) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@adiprerepa: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 58f508587be7eab1419862ba4f5df26f4d8e05b9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/lint_istio/15654) | `/test lint_istio`
gencheck_istio | 58f508587be7eab1419862ba4f5df26f4d8e05b9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/gencheck_istio/14249) | `/test gencheck_istio`
unit-tests_istio | 58f508587be7eab1419862ba4f5df26f4d8e05b9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/unit-tests_istio/16249) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

lambdai committed:
networking: add better transport security iptables rules (#25392)

* add better transport security iptables rules

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

* make lint

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

howardjohn committed:
Add targets for running/reporting/comparing benchmarks as part of CI (#25257)

* Add targets for run/compare/report

* fix lint

ostromart committed:
Refactor some more tests away from using goldens (#25405)

linsun committed:
Allow users to enable validation webhook or not (#25372)

* enable validation

* update

* use existing flag John pointed out

* Update values.yaml

Commit:
Update helm v3 dependency (#25407)

ramaraochavali committed:
remove version check for tcp metadata filter (#25354)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

howardjohn committed:
Fix crd client metrics (#25376)

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#25409)

howardjohn committed:
Handle tombstone in ledger updates (#25377)

* Handle tombstone in ledger updates

* lint

bison committed:
Fix tombstone recovery bug in Galley (#25358)

esnible committed:
Check for cert expiration locally before contacting server (#25314)

Commit:
Use MUTUAL_TLS for ControlPlaneAuthPolicy in default ProxyConfig (#25411)

GLYASAI committed:
Make proxyv2 image building operations universal (#25146)

* fix deployment.yaml for prometheus.

* fix typo.

* use SIDECAR instead of envoy

* change X_ENVOY_X  to X_SIDECAR_X

* Use envoy as the sidecar by default

* change envoy to sidecar

* change envoy to sidecar

* make lint happy

* copy *-filter.wasm

* compatible with XXX_ENVOY_XXX variables

* make lint happy

* missed code

Co-authored-by: sky <fakangwang@gmail.com>
Co-authored-by: Jimmy Song <rootsongjc@gmail.com>

howardjohn committed:
Fix telemetry benchmark (#25275)

This will run in CI soon, so it won't break

stevenctl committed:
sidecar scopes pushes to namespace (#25403)

* sidecar updates scoped to namespace

* include root namespace

* check namespaces match

* test framework allows getting istiod stats

* add integration test

* ConvertToSidecarScope keeps track of root ns

* set root ns in test

* sidecar change does not trigger gateway push

* remove integration test

* format

* keep checking configs

stevenctl committed:
optimize CombineVHostRoutes (#25378)

* wait to sort catchAll until all appends finish

* oneshot append and sort

* Revert "oneshot append and sort"

This reverts commit 87273a7476dde70147055eaa7c5b4445c2e0aa0b.

* consolidate Combine and Sort VHostRoutes

howardjohn committed:
Bump service-apis dependency (#25390)

howardjohn committed:
Revert "Add MultiVM to VM tests  (#25341)" (#25421)

This reverts commit 7be5b40aab004c239624df791e49e00640655e28.

howardjohn committed:
Fix SHA reference (#25422)

The old one is `master`, we want the SHA.

howardjohn committed:
Add roundtrip fuzzing tests for our CRDs (#25290)

* Add roundtrip fuzzing tests for our CRDs

For https://github.com/istio/istio/issues/14935

I explicitly tested this would have caught two bugs we have had around
CRD pruning (one currently blocking istio/api update)

* fix lint

* fix lint

howardjohn committed:
Add initial RPM build and test (#25316)

* Add initial RPM build and test

* fixes

* Add _8 suffix

* Revert defaults back to debian

howardjohn committed:
Update grafana and prometheus (#25365)

xulingqing committed:
Add unit test for mode/authentication to cover ConstructSdsSecretConfigWithCustomUds, ConstructValidationContext, ApplyCustomSDSToCommonTLSContext (#25306)

* add unit test for authz and fix lint issue

* fix comments

* fix comments

* resolve comments

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#25431)

nmittler committed:
Moving security envoy filters to pkg/envoy (#25429)

Also adding the istio-specific filters to the generated import list.

brian-avery committed:
Added template for release notes (#25368)

* Added template for release notes

* Update template.yaml

* Update template.yaml

* Update template.yaml

* Update CODEOWNERS

* Update template.yaml

* Update CODEOWNERS

howardjohn committed:
Add memory to benchmark test (#25435)

nschhina committed:
[Bug Fix] Empty ResouceName results in ill configured UpstreamTLSContext (#25433)

* fix bug

* check mutual mode as well

* address concerns

nmittler committed:
Expose more of istioctl as library (#25384)

This is another step toward removing the istioctl test component.

therealmitchconnors committed:
record reverseStatus as hashset, due to high cpu utilization (#24923)

* record reverseStatus as hashset, due to high cpu utilization

* fix merge pasta

istio-testing commented:
@adiprerepa: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

knight42 committed:
fix(pilot): apply header manipulations in redirect action (#25444)

nmittler committed:
Fix breakage in istioctl manifest apply (#25451)

This was introduced by #25384. The issue was that the parameter order was swapped in order to be consistent with GenerateConfig (and a few other places). There was apparently no test coverage for this beyond the integration test framework's usage when installing istio. Unfortunately, we don't want to use the commands directly there because the use of flags interferes with the ability to install multiple control planes in parallel.

Fixing the issue by swapping to a structure for the parameter to avoid similar breakage in the future.

Fixes #25447

tbarrella committed:
Rename AuthWhiteList and AuthBlackList variables (#25439)

* Rename AuthWhiteList/AuthBlackList variables

As part of #25381

This is just s/g. Updating the comments, etc. comes next

* Update strings/comments around AuthAllowlist and AuthDenylist

As part of #25381

The contents of `ListAuthChecker.String()` aren't tested, making this
seem safe

knight42 committed:
refactor(pilot): extract method to get header operations (#25455)

Signed-off-by: knight42 <anonymousknight96@gmail.com>

hzxuzhonghu committed:
use service lister (#25457)

esnible committed:
Don't complain about unknown keys on in-cluster IOP; --manifests for 'istioctl verify-install' (#25452)

ramaraochavali committed:
update endpoint shards when doing a full push for service entries (#25414)

* wip fix

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* revert

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix test cases

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix event delete

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix race

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix lock

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* remove unnecessary test locks

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* add unchanged svcs

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* move mutex to memory store

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

ramaraochavali committed:
wip (#25459)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#25469)

howardjohn committed:
Revert "wip (#25459)" (#25473)

This reverts commit a736986dd3b7523300a9bc611d991c586266aa11.

nmittler committed:
Revert "Expose more of istioctl as library (#25384)" (#25471)

This reverts commit 5ef2bdd5

stevenctl committed:
cleanup ingress test component (#25475)

* tests: shorten massive log line from ingress component

* another long log

* DRY getHTTPSAddressInner

* format

howardjohn committed:
Use XDS v3 for SDS (#25335)

* Use XDS v3 for SDS

For https://github.com/istio/istio/issues/19885

* fix lint

* remove debug

* wip

* fix test

* more tests

brian-avery committed:
Add api version to release notes schema (#25466)

* Update template.yaml

* Update template.yaml

irisdingbj committed:
add config map and envoy filter into istiod-remote (#25404)

* add istio cm and envoy filter into istiod-remote

* update values

knight42 committed:
test(security): deflake TestCreatePluggedCertCA (#25446)

Signed-off-by: knight42 <anonymousknight96@gmail.com>

Commit:
Cleaned proxy config regression (#25460)

howardjohn committed:
Tune benchmark test (#25472)

Remove the million endpoint test (we are so far way from scaling that
high it doesn't matter really) because it constantly leads to CPU
throttling both in CI and locally which leads to inaccurate results.

Increase count to 5 per recommendation of
https://github.com/golang/go/issues/23471

dgn committed:
manifests: fix injection of ProxyMetadata into init container (#25474)

Previously, we would only inject ProxyMetadata if
proxy_init.resources was set.

tbarrella committed:
Feature labels test whitelist -> allowlist (#25476)

As part of #25381

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#25481)

stevenctl committed:
tests: remove pilot from echo config (#25279)

* tests: remove pilot from echo config

* cleanup multicluster usgaes

* outboundtrafficpolicy fix

* remove pilot param from EchoConfig helper

* fix new vm tests

ramaraochavali committed:
add gateway server name as stat prefix (#25456)

* add gateway server name as stat prefix

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* lint

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

ramaraochavali committed:
ads: fix adsc version nonce handling (#25418)

* wip

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix adsc

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* remove unnecessary code

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix tests

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* remove unnecessary print

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix ack

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix test

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

ramaraochavali committed:
Reverts Revert of "gateway: do not compute snihosts for service repeatedly" (#25488)

* wip

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* lint

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

hanxiaop committed:
Fix a bug for an error message type in "istioctl analyze" feature (#25489)

* Fix a bug for an error message in  func

* make gen to pass gen test

* Update messages.gen.go

* change messages.yaml and make gen

* Update messages.gen.go

Commit:
Split mirrors into separate path in virtualservice analyzers (#25442)

howardjohn committed:
Increase timeout for benchtest (#25487)

* Increase timeout for benchtest

Sometimes it goes too long:
https://storage.googleapis.com/istio-prow/logs/benchmark-report_istio_postsubmit/40/artifacts/benchmark-log.txt

* fix lint

howardjohn committed:
Move validation webhook to shared informers (#25130)

Commit:
Add script for updating the proxy SHA for use by automation (#25496)

hzxuzhonghu committed:
Do not enable auto mtls when cluster type is `Cluster_ORIGINAL_DST` (#24319)

* disable tls for headless service when mtls is auto detected

* move it to conditionallyConvertToIstioMtls

* cleanup

* fix e2e

irisdingbj committed:
support remote KUBECONFIG for istiod (#25068)

ramaraochavali committed:
router: do not send deprecated max program size to 1.7 proxies (#25492)

* max program size

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* lint

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

howardjohn committed:
Trigger endpoint update when pod comes in (#25161)

* Trigger endpoint update when pod comes in

* Add testing and endpoint slice

* Fix stale comment

* Fix locks

* comments and metric

* Remove lint error

costinm committed:
Refactor interfaces that are used in pkg, to reduce copying and deps (#25367)

* Refactor interfaces that are used in pkg, to reduce copying and deps

* Move SecretItem as well, last dep

* Even if using file mounted certs, we still want SDS

* More fixes

* More fixes

* Fix merge

* Missed the most important file

* Few more merge errors

* Format and line too long

* Format and line too long

* revert change to go.mod

* Accidentally renamed the env variable name

stevenctl committed:
repeatably run integ-suite-kind (#25470)

* repeatably run integ-suit-kind

* fill in vars

* add newline to yaml

* base mc kind config off trustworthy-jwt.yaml

* lint

tbarrella committed:
Replace misc. occurrences of whitelist/blacklist (#25479)

As part of #25381

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#25497)

howardjohn committed:
Bump release-builder dep (#25501)

istio-testing committed:
Automator: update istio/api@master dependency in istio/istio@master (#25251)

stevenctl committed:
echo retries GetRemoteDiscoveryAddress (#25499)

* echo retries GetRemoteDiscoveryAddress

* format

esnible committed:
More output columns for 'istioctl pc routes' and 'clusters' (#24867)

* More output columns for 'istioctl pc routes'

* Suppress passthrough routes.  Add DR column to 'pc clusters'

* Rename DR and VS columns

* Remove CONFIG column

Commit:
gateway charts should not depend on global.yaml (#25402)

Fix: Missing meshConfig for both ingress and egress

Commit:
Fix egress gateway config. (#25508)

jasonwzm committed:
Move CRDs to v1 (#25263)

* Move CRDs to v1

* make gen

* update api

* gomod

* fix some galley tests

* fix mixer tests

* fix conflict

ostromart committed:
Run generate tests also through istioctl install and operator (#25254)

* Run tests through kubebuilder test env

* Fix controller and apply cases

* Fix channel problem due to global use

* Cleanup

* Add comment

* Lint

* Refresh goldens

* Remove copyright year

* Update comments

* Add comment to prune list

howardjohn committed:
namespace controller: use shared informer (#25123)

* Update namespace controller to use shared informer

* Fix leader election

* rebase

hzxuzhonghu committed:
For k8s registry the instance handler is redudant (#25462)

* For k8s registry the instance handler is redudant

* update comment

Commit:
Remove RequireCerts from SDS Agent (#25458)

ramaraochavali committed:
minor ads renaming (#25537)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

ramaraochavali committed:
disable enforcing successrate based outlier detection (#25534)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#25540)

ramaraochavali committed:
refactor pod lookup code (#25524)

* refactor pod lookup code

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* lint fixes

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

Commit:
optimize memory usage (#25531) (#25532)

tbarrella committed:
Use MARKDOWN_LINT_ALLOWLIST instead of _WHITELIST (#25545)

Following https://github.com/istio/common-files/pull/276

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#25547)

justinwei2 committed:
Azure Platform Support (#24995)

* azure rough template

* azure interface with API version update

* fixed lints

* moved API version update inside azureEnv

* added extra metadata, ignore empty fields

* make metadata parsers azureEnv recievers

* added azure_ prefix to tags

Co-authored-by: Justin Wei <juswei@google.com>

esnible committed:
XDS-based replacement for /debug/syncz (#25344)

* Experimental XDS-based replacement for /debug/syncz

* istioctl client side implementation for XDS istio.io/debug/syncz

* Handle 'x proxy-status <pod>'

* Early out 'proxy-status <pod>', pass proxyID as resource name

* Initial error handling

* Logging and output error handling

* Clean-up

howardjohn committed:
Optimize DR lookup in EDS (#25518)

* Improve EDS benchmark

* Optimize DR lookup in EDS

```
name                         old time/op    new time/op    delta
EndpointGeneration/1/100-8      257µs ±16%     157µs ±11%  -39.00%  (p=0.008 n=5+5)
EndpointGeneration/10/10-8      100µs ± 9%     102µs ±14%     ~     (p=0.841 n=5+5)
EndpointGeneration/100/10-8     590µs ±13%     626µs ±11%     ~     (p=0.310 n=5+5)
EndpointGeneration/1000/1-8     869µs ± 8%     940µs ± 7%     ~     (p=0.095 n=5+5)

name                         old alloc/op   new alloc/op   delta
EndpointGeneration/1/100-8     75.1kB ± 0%    75.1kB ± 0%     ~     (p=0.357 n=5+5)
EndpointGeneration/10/10-8     22.2kB ± 0%    22.2kB ± 0%     ~     (p=0.095 n=5+5)
EndpointGeneration/100/10-8    41.9kB ± 0%    42.0kB ± 0%     ~     (p=0.730 n=5+5)
EndpointGeneration/1000/1-8    37.7kB ± 3%    37.9kB ± 1%     ~     (p=0.690 n=5+5)

name                         old allocs/op  new allocs/op  delta
EndpointGeneration/1/100-8      1.30k ± 0%     1.30k ± 0%     ~     (all equal)
EndpointGeneration/10/10-8        395 ± 0%       395 ± 0%     ~     (all equal)
EndpointGeneration/100/10-8       401 ± 1%       401 ± 0%     ~     (p=0.730 n=5+5)
EndpointGeneration/1000/1-8      62.0 ± 0%      62.0 ± 0%     ~     (all equal)
```

* fix tests

* fix lint

* fix test

* Add release note

* wrong pr

* refactor

* fix nil

howardjohn committed:
Remove inaccurate and non-performant EDS metrics (#25519)

* Remove inaccurate and non-performant EDS metrics

```
name                           old time/op    new time/op    delta
EndpointGeneration/100/1-8       94.3µs ± 7%    89.7µs ±17%     ~     (p=0.635 n=3+3)
EndpointGeneration/1000/1-8       855µs ± 2%     822µs ± 3%     ~     (p=0.125 n=3+3)
EndpointGeneration/10000/1-8     7.62ms ± 6%    7.14ms ± 3%     ~     (p=0.162 n=3+3)
EndpointGeneration/100/100-8      714µs ± 3%     665µs ± 7%     ~     (p=0.156 n=3+3)
EndpointGeneration/1000/100-8    5.17ms ± 7%    6.45ms ±20%     ~     (p=0.193 n=3+3)

name                           old alloc/op   new alloc/op   delta
EndpointGeneration/100/1-8       6.59kB ± 0%    4.69kB ± 0%  -28.90%  (p=0.000 n=3+3)
EndpointGeneration/1000/1-8      37.9kB ± 0%    35.9kB ± 0%   -5.29%  (p=0.000 n=3+3)
EndpointGeneration/10000/1-8      312kB ± 5%     304kB ± 3%     ~     (p=0.434 n=3+3)
EndpointGeneration/100/100-8     89.9kB ± 0%    78.5kB ± 0%  -12.76%  (p=0.000 n=3+3)
EndpointGeneration/1000/100-8     569kB ± 4%     533kB ± 9%     ~     (p=0.310 n=3+3)

name                           old allocs/op  new allocs/op  delta
EndpointGeneration/100/1-8         60.7 ± 1%      21.0 ± 0%  -65.38%  (p=0.000 n=3+3)
EndpointGeneration/1000/1-8        60.7 ± 1%      21.0 ± 0%  -65.38%  (p=0.000 n=3+3)
EndpointGeneration/10000/1-8       63.3 ± 3%      29.0 ± 0%  -54.21%  (p=0.001 n=3+3)
EndpointGeneration/100/100-8      1.20k ± 0%     0.96k ± 0%     ~     (zero variance)
EndpointGeneration/1000/100-8     4.38k ± 6%     3.90k ±14%     ~     (p=0.225 n=3+3)
```

These metrics are both extremely costly and are actually not accurate. The metric tries to give a global view of the number of endpoints in a Service - this is wrong by definition, as each proxy will have a different view of EDS based on sidecar/network.

* add release note

* Update releasenotes/25519.yaml

Co-authored-by: Brian Avery <bavery@redhat.com>

Co-authored-by: Brian Avery <bavery@redhat.com>

nmittler committed:
[Test Framework] Reduce use on Environment (#25522)

This is additional cleanup to remove more direct use of the Environment API by tests and throughout the framework.

nschhina committed:
Disable SDS Config generation for Sidecar Proxy when CredentialName is set (#25517)

* fix incorrect SDS config generation

* make gen

howardjohn committed:
Pin number of CPUs for benchmark (#25552)

Right now its using 16 CPUs but is limited to 8. I think this leads to
flaky results

howardjohn committed:
Handle foreign instances by workload rather than endpoint (#25502)

* Handle foreign instances by workload rather than endpoint

Currently foreign instances are based on ServiceInstance. This has two
issues:

* They are derived from Endpoint, which has the Service -> Port mapping
already done, breaking the port fields in ServiceEntry
* They are derived from Endpoint, meaning you need a Service AND
ServiceEntry to select a pod
* They are basically hacking the ServiceInstance by passing a bunch of
bogus fields in places that are relevant and then later overwriting
them, rather than passing the info we actually know and letter the
handlers work only on that info

* format

* Naming

* fix race condition

tariq1890 committed:
pin a commit SHA of openshift/api as a dep instead (#25486)

Commit:
Add env to support RSA Key size for Istio self signed CA certificates (#25188)

* Add env to support specification of CA generated certificates

When user specifies CITADEL_CA_RSA_KEY_SIZE that key size is used to
generate Istio CA certificates allowing for finer control of security.

* Fix unit test

* Run make gen

rshriram committed:
fix target ports in workload entries (#25428)

* fix target ports in workload entries

Signed-off-by: Shriram Rajagopalan <rshriram@tetrate.io>

* lint

Signed-off-by: Shriram Rajagopalan <rshriram@tetrate.io>

* nits

* make gen

Signed-off-by: Shriram Rajagopalan <rshriram@tetrate.io>

* Revert "make gen"

This reverts commit 79ad9ca53ef269bb4498eeac2f3c5f36e8864654.

* revert go mod

* no 9090

Signed-off-by: Shriram Rajagopalan <rshriram@tetrate.io>

* fix headless target port

Signed-off-by: Shriram Rajagopalan <rshriram@tetrate.io>

* lint

Signed-off-by: Shriram Rajagopalan <rshriram@tetrate.io>

* undo security change

cagataygurturk committed:
Properly quote .Values.telemetry.v2.accessLogPolicy.logWindowDuration (#25551)

* Properly quote .Values.telemetry.v2.accessLogPolicy.logWindowDuration value

Fixes #25550

* Add generated files

* Run make gen

stevenctl committed:
test analyzer does not fail unlabeled tests when suite is skipped (#25427)

* test analyzer allows skipped tests without feature

* add target for analyzing specific suites

daixiang0 committed:
fix wrong comment (#25563)

Signed-off-by: Xiang Dai <long0dai@foxmail.com>

douglas-reid committed:
feat(testing): Add VM-based Stackdriver tests for metrics and logs. (#25340)

* Add VM-based Stackdriver tests for metrics and logs.

* use default vm image

* add feature for stackdriver telemetry

* update from rebase

Commit:
Specify helm keys and values that are cross-referenced by other Istio helm charts (#25569)

* Important values that affect multiple charts should be called out explicitly

* Fix egress charts

* Fix discovery chart

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#25583)

howardjohn committed:
echo: sort headers (#25582)

* echo: sort headers

Currently the response bounces around a lot as header order is not
defined which makes debugging annoying sometimes

* format

brian-avery committed:
Add release note for adding release notes process (#25509)

* Add release note for adding release notes process

* Add readme

* Fix readme

* Fix linter errors

* Updates based on PR comments

* Moved 25519 into the notes directory

* Added 25519.yaml

* Updated 25519.yaml to v2

* Update add-release-notes-generation.yaml

* Update add-release-notes-generation.yaml

justinwei2 committed:
GCP instance labels support (#24687)

* added gcp labels support

* labels are only retrieved for gce vm instances, changed timeout behavior

* handle no instance labels case

* don't pull labels if running on kubernetes

* cleaned up platform interface

* cache metadata inside gcpEnv

* Revert "cache metadata inside gcpEnv"

This reverts commit 361ff7fbaa609c62af4987b708890ff068899aef.

* restructured cache metadata inside gcpEnv

* minor fixes

Co-authored-by: Justin Wei <juswei@google.com>

lambdai committed:
fix dns rules on v4 only system (#25590)

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

istio-testing committed:
Automator: update common-files@master in istio/istio@master (#25593)

richardwxn committed:
add uninstall by revision change (#25076)

* add uninstall change

* add test

* add filename flag, refactor

* refactor, move common function from cmd/mesh to pkg

* clean up the pruning and deleting approaches

* address comment

* address comments

* add --purge option and prune_test

* clean up code and add tests

* address more comments

* update label logic

* rebase

* update label logic

* address comments and fix lint

* fix test

bianpengyuan committed:
Update proxy SHA (#25594)

* update istio proxy sha

* fix test

* lint

luksa committed:
Allow users to delay application start until proxy is ready (#24737)

nschhina committed:
Reenable GatewayTLSOrigination Test (#25424)

* renable test

* switch back ports

* fix test

* tests pass locally

* make gen

* add subset for route to gateway

* add sni

* change timeout

* Update egress_gateway_origination_test.go

* split virtual service

* add more wait

* more wait

* I think this works

* decrease sleep time

* remove sleep and increase timeout

ramaraochavali committed:
rename foreign instances to workload instances (#25605)

* rename foreign instances to workload instances

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix race

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

brian-avery committed:
Update template.yaml (#25608)

esnible committed:
Config file and env var for istioctl --istioNamespace, --xds-address, and --cert-dir (#25280)

* Config file and env var for some istioctl CLI options

* ISTIOCONFIG variable for overriding default istioctl configuration

* 'prefer-experimental' option for commands with regular and experimental variants

* Defaults for --xds-san and --insecure

* Column for 'istioctl x config list' that lets users tell defaults from configured values

* Fail if user supplies invalid ISTIOCONFIG env var

* Don't fail if config file does not exist

* Initialize defaults in tests

* Show origin of config var; move defaulting close to command so tests work

* Use Istio RegisterXXXVar for environment overrides to istioctl

* Allow user to make XDS-based proxy-status the default with env or config setting

* Added release note

* Sort 'x config list' output

* release note fix

GregHanson committed:
Egress Gateway TLS Origination fix (#25588)

* add tlsMode=istio label to egress gateway

* Revert "add tlsMode=istio label to egress gateway"

This reverts commit a8310ddf0b21db928abbf83a5df34194ab76bd48.

* clean out TransportSocketMatches on subsequent applyTrafficPolicy calls

stevenctl committed:
sidecar scope matches ns for envoy filter and authn/z (#25430)

* sidecar scope matches ns for envoy filter and authn/z

* format

* remove peer authn

howardjohn committed:
Optimize memory usage of SDS cluster config generation (#25511)

* Add TLS to bench tests

* Apply optimizations for SDS generation

* Fix tests

* fix test

adiprerepa committed:
rebase

adiprerepa commented:
okay, rebase screwed this up somehow. moving PR.

istio-testing commented:
@adiprerepa: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 58f508587be7eab1419862ba4f5df26f4d8e05b9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/lint_istio/15654) | `/test lint_istio`
gencheck_istio | 58f508587be7eab1419862ba4f5df26f4d8e05b9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/gencheck_istio/14249) | `/test gencheck_istio`
unit-tests_istio | 58f508587be7eab1419862ba4f5df26f4d8e05b9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/unit-tests_istio/16249) | `/test unit-tests_istio`
integ-mixer-k8s-tests_istio | 3413e0c5d15894a9af04bbfe4ca40773dd342358 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25389/integ-mixer-k8s-tests_istio/14762) | `/test integ-mixer-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->
